test/boost/aws_error_injection_test.cc

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "test/lib/log.hh"
+#include "test/lib/random_utils.hh"
+#include "test/lib/scylla_test_case.hh"
+#include "test/lib/tmpdir.hh"
+#include "utils/exceptions.hh"
+#include "utils/s3/aws_errors.hh"
+#include "utils/s3/client.hh"
+#include <seastar/core/fstream.hh>
+#include <seastar/core/units.hh>
+#include <seastar/http/httpd.hh>
+#include <seastar/util/closeable.hh>
+
+using namespace seastar;
+using namespace std::chrono_literals;
+struct server {
+    enum class failure_policy : uint8_t {
+        SUCCESS = 0,
+        RETRYABLE_FAILURE = 1,
+        NONRETRYABLE_FAILURE = 2,
+    };
+    class dummy_aws_request_handler : public httpd::handler_base {
+    public:
+        explicit dummy_aws_request_handler(server& test_server) : _test_server(test_server) {}
+        future<std::unique_ptr<http::reply>> handle(const sstring& path, std::unique_ptr<http::request> req, std::unique_ptr<http::reply> rep) override {
+            auto method = req->_method;
+            auto url_params = req->query_parameters;
+            testlog.debug("{}\t{}", req->_method, req->get_url());
+            rep->write_body("txt", build_response(*req));
+            if (method == "DELETE") {
+                rep->set_status(http::reply::status_type::no_content);
+            } else if (method == "PUT") {
+                rep->set_status(http::reply::status_type::ok);
+                if (url_params.contains("partNumber") && url_params.contains("uploadId")) {
+                    rep->add_header("ETag", "SomeTag_" + url_params.at("partNumber"));
+                }
+                rep->add_header("ETag", "SomeTag");
+            }
+            return make_ready_future<std::unique_ptr<http::reply>>(std::move(rep));
+        }
+
+    private:
+        sstring build_response(const http::request& req) {
+            sstring response;
+            if (req._method == "POST") {
+                if (req.query_parameters.contains("uploads")) {
+                    response = "<InitiateMultipartUploadResult><Bucket>bucket</Bucket><Key>key</Key><UploadId>UploadId</UploadId></"
```

Comment:
why not just `return R"..."`  directly? if we don't expect other HTTP methods, i'd suggest:
```c++
assert(req._method == "POST");
```

Reply:
Done